libcloud/dns/drivers/cloudflare.py

```diff
@@ -141,290 +159,252 @@ class CloudFlareDNSDriver(DNSDriver):
+        item = response.object['result']
+        zone = self._to_zone(item)
 
-    def ex_set_zone_security_level(self, zone, level):
-        """
-        Set the zone Basic Security Level to I'M UNDER ATTACK! / HIGH /
-        MEDIUM / LOW / ESSENTIALLY OFF.
+        return zone
 
-        :param level: Security level. Valid values are: help, high, med, low,
-                      eoff.
-        :type level: ``str``
+    def update_zone(self, zone, domain, type='master', ttl=None, extra=None):
         """
-        params = {'z': zone.domain, 'v': level}
-        self.connection.set_context({'zone_domain': zone.domain})
-        resp = self.connection.request(action='sec_lvl', params=params)
-        result = resp.object
-        return result.get('result', None) == 'success'
+        @inherits: :class:`DNSDriver.update_zone`
 
-    def ex_set_zone_cache_level(self, zone, level):
+        Note that the ``zone``, ``type`` and ``ttl`` properties can't be
+        updated. The only updatable properties are the ``extra`` zone
+        properties specified in ``ZONE_UPDATE_ATTRIBUTES``. Only one property
+        may be updated at a time. Any non-updatable properties are ignored.
         """
-        Set the zone caching level.
+        body = merge_valid_keys({}, ZONE_UPDATE_ATTRIBUTES, extra)
+        if len(body) != 1:
+            return zone
 
-        :param level: Caching level. Valid values are: agg (aggresive), basic.
-        :type level: ``str``
-        """
-        params = {'z': zone.domain, 'v': level}
-        self.connection.set_context({'zone_domain': zone.domain})
-        resp = self.connection.request(action='cache_lvl', params=params)
-        result = resp.object
-        return result.get('result', None) == 'success'
+        url = '{}/zones/{}'.format(API_BASE, zone.id)
 
-    def ex_enable_development_mode(self, zone):
-        """
-        Enable development mode. When Development Mode is on the cache is
-        bypassed. Development mode remains on for 3 hours or until when it is
-        toggled back off.
-        """
-        params = {'z': zone.domain, 'v': 1}
-        self.connection.set_context({'zone_domain': zone.domain})
-        resp = self.connection.request(action='devmode', params=params)
-        result = resp.object
-        return result.get('result', None) == 'success'
+        self.connection.set_context({'zone_id': zone.id})
+        response = self.connection.request(url, data=body, method='PATCH')
 
-    def ex_disable_development_mode(self, zone):
-        """
-        Disable development mode.
-        """
-        params = {'z': zone.domain, 'v': 0}
-        self.connection.set_context({'zone_domain': zone.domain})
-        resp = self.connection.request(action='devmode', params=params)
-        result = resp.object
-        return result.get('result', None) == 'success'
+        item = response.object['result']
+        zone = self._to_zone(item)
 
-    def ex_purge_cached_files(self, zone):
```

Comment:
It looks like some of those extension methods haven't been implemented in the v4 driver.

Not a big deal for now, but eventually it would be good to implement the ones which are still supported by v4 API.

Reply:
I've added placeholders for the extension methods in 48330b7 so that we don't loose track of them. Given that currently the driver is fully non-functional and given that there are quite a few of the extension methods, would you be comfortable with getting the basic API merged and add the extension methods as they are requested by users and/or more development bandwidth frees up?

Reply:
Yeah, I agree, not a blocker.